src/engine/enginemixer.h

```diff
@@ -336,6 +336,7 @@ class EngineMixer : public QObject, public AudioSource {
     CSAMPLE_GAIN m_headphoneGainOld;
     CSAMPLE_GAIN m_balleftOld;
     CSAMPLE_GAIN m_balrightOld;
+    std::atomic<unsigned int> m_numMicsConfigured;
```

Comment:
Did you check that this is needed due to concurrent access or was this just a safe guess?

Reply:
`EngineMixer::process` reads the CO in the engine thread and I think the `onInputConnected`/`onInputDisconnected` methods that modify the value are likely triggered from the GUI thread (although I didn't check).